Adds an Apache Guacamole guide (#1547)
pomerium/documentation: content/docs/guides/guacamole.md

---
# cSpell:ignore guacd, GUACD, guacnetwork
title: Secure Apache Guacamole
description: This guide shows you to secure Apache Guacamole behind Pomerium using JWT Claims Headers to support HTTP header authentication.
keywords: [JWT claims headers, http auth, header auth]
sidebar_label: Apache Guacamole
---

# Secure Apache Guacamole

> This guide shows you how to secure an instance of [Apache Guacamole](https://guacamole.apache.org/), a clientless, remote desktop gateway that supports VNC, RDP, and SSH protocols.

Reviewer: Lets clarify the objective at the start &  explicitly state the goal in the introduction, mentioning that the guide will help users set up a secure, authenticated Apache Guacamole instance behind Pomerium.

Author: I rewrote the intro based on your feedback.

> This guide shows you how to secure an instance of [Apache Guacamole](https://guacamole.apache.org/), a clientless, remote desktop gateway that supports VNC, RDP, and SSH protocols.

Reviewer: ```suggestion
This guide will walk you through securing an instance of [Apache Guacamole](https://guacamole.apache.org/) behind Pomerium, leveraging JWT Claims Headers for HTTP header authentication. The setup allows centralized, secure, and seamless authentication via Pomerium. [Apache Guacamole](https://guacamole.apache.org/), a clientless, remote desktop gateway that supports VNC, RDP, and SSH protocols.
```

> 

Reviewer: ```suggestion
+To complete this guide, you will:
+1. Install and configure Guacamole using Docker Compose.
+2. Replace the default Nginx with Pomerium in the Docker setup.
+3. Configure Pomerium to forward identity headers to Guacamole.
+4. Securely run your setup and verify proper authentication.
```

:::note

This guide provides steps to secure access to the Guacamole gateway behind Pomerium. It does not cover how to add connections to Guacamole itself.

:::

### Before you start

To complete this guide, you need:

- [Pomerium Core](/docs/core)
- [Docker](https://docs.docker.com/install/) and [Docker Compose](https://docs.docker.com/compose/install/)

:::note

This guide uses the [Hosted Authenticate Service](/docs/capabilities/hosted-authenticate-service.md), which provides its own hosted identity provider and authenticate service URL. If you prefer to configure your own identity provider, see [Self-Hosted Authenticate Service](/docs/capabilities/self-hosted-authenticate-service).

:::

## Install Guacamole with Docker Compose

[Installing Guacamole](https://guacamole.apache.org/doc/gug/guacamole-docker.html) with Docker requires the following services:

- `guacamole/guacd`: The guacd daemon, which provides support for VNC, RDP, SSH, telnet, and Kubernetes.
- `guacamole/guacamole`: The Guacamole web application running within a Tomcat 8 server with websocket support.
- `mysql` or `postgres`: The relational database that stores authentication and connection configuration data.

> 

Reviewer: ```suggestion
> **Security Best Practice:** Ensure that environment variables containing sensitive data, such as passwords, are stored securely. Consider using Docker secrets or an `.env` file.

```

     ports:
       - 8443:443
     links:
       - guacamole
     networks: guacnetwork_compose
   ```
1. Add the Pomerium Core configuration:

   ```yaml showLineNumbers
   pomerium:
     image: cr.pomerium.com/pomerium/pomerium:latest
     # highlight-start
     # Mount your config file
     volumes:
       - ./config.yaml:/pomerium/config.yaml:ro
     # highlight-end
     ports:
       - 443:443
     networks:
       - guacnetwork_compose
     environment:
       JWT_CLAIMS_HEADERS: email
   ```

1. In line 5 above, you mount a Pomerium configuration file into Docker Compose. In your project's root directory, create a `config.yaml` file with the following configuration:

   ```yaml title="Pomerium configuration file"
   authenticate_service_url: https://authenticate.pomerium.app

   jwt_claim_headers: email

   routes:
     - from: https://guacamole.localhost.pomerium.io
       to: http://guacamole:8080
       policy:
         - allow:
             or:
               - email:
                   # highlight-next-line
                   is: user@example.com
       pass_identity_headers: true
   ```

   Don't forget to replace `user@example.com` with your own email address.

> 

Reviewer: Can you show the full working configurations / yamls at the end?

Author: Yes, I'll add a section to do that.
